apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/SpannerDate.cs

```diff
@@ -0,0 +1,279 @@
+        {
+            string message = TryParseImpl(text, out var value);
+            return message == null ? value : throw new FormatException(message);
+        }
+
+        /// <summary>
+        /// Attempts to parse a textual representation of a <see cref="SpannerDate"/> value in <see cref="CultureInfo.InvariantCulture"/> format.
+        /// </summary>
+        /// <remarks>
+        /// See <see cref="Parse(string)"/> for format details. This method will return <c>true</c> if and only if
+        /// <see cref="Parse(string)"/> would return without an exception.
+        /// </remarks>
+        /// <param name="text">The text to parse. Must not be null.</param>
+        /// <param name="value">The parsed value, or 0 on failure.</param>
+        /// <returns><c>true</c> if <paramref name="text"/> was parsed successfully; <c>false</c> otherwise.</returns>
+        public static bool TryParse(string text, out SpannerDate value)
+        {
+            string message = TryParseImpl(text, out value);
+            return message == null;
+        }
+
+        /// <summary>
+        /// Returns the error message for a FormatException if parsing fails, or null for success.
+        /// </summary>
+        private static string TryParseImpl(string text, out SpannerDate value)
+        {
+            GaxPreconditions.CheckNotNull(text, nameof(text));
+
+            if (DateTime.TryParse(text, CultureInfo.InvariantCulture, DateTimeStyles.None, out DateTime dateTime))
+            {
+                value = new SpannerDate(dateTime);
+                return null;
+            }
+            else
+            {
+                value = default;
+                return "Text representation must be a valid date.";
+            }
+        }
+
+        /// <summary>
+        /// Returns the ISO 8601 representation (yyyy-MM-dd) of the date represented by this instance.
+        /// </summary>
+        /// <returns>The ISO 8601 string representation of the date represented by this instance.</returns>
+        public override string ToString() => ToDateTime().ToString("yyyy-MM-dd", CultureInfo.InvariantCulture);
+
+        /// <summary>
+        /// Returns a SpannerDate instance that is set to the date part of the specified DateTime.
+        /// </summary>
+        /// <param name="dateTime">The date time.</param>
+        /// <returns>The SpannerDate instance</returns>
+        public static SpannerDate FromDateTime(DateTime dateTime) => new SpannerDate(dateTime);
+
+        /// <summary>
+        /// Converts the value of this SpannerDate instance to the <see cref="DateTime"/> of kind <see cref="DateTimeKind.Unspecified"/>.
+        /// </summary>
+        /// <returns>The DateTime value.</returns>
+        public DateTime ToDateTime() => DateTime.MinValue.AddDays(_days);
+
+        /// <summary>
+        /// Returns a new <see cref="SpannerDate"/> that adds the specified number of days to the value of this instance.
```

Comment:
Possibly highlight that the existing instance is not modified by this method.
(I can see the consistency reasons for calling it AddDays... in Noda Time we'd call this PlusDays to make it clearer :)